### PR TITLE
[IMP] l10n_ca: Add/Imp COA, taxes, reports

### PR DESCRIPTION
**Pull request by JulienAlardot**
Complete refactor of the module,
adding the tax report

Task: task-3098316

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_ca/models/template_ca.py [hunk 1]
```diff
@@ -24,13 +24,44 @@ def _get_ca_res_company(self):
...
+                domain += [('amount_type', '=', 'group'), ('amount', '=', 100), ('description', '=', '5% GST + 9.975% QST')]
+            else:
+                domain += [('amount_type', '=', 'percent'), ('amount', '=', 5)]
+
+            sales_domain = domain + [('type_tax_use', 'in', ('sale', 'all'))]
+            purchases_domain = domain + [('type_tax_use', 'in', ('purchase', 'all'))]
+            return self.env['account.tax'].search(sales_domain, limit=1).id, self.env['account.tax'].search(purchases_domain, limit=1).id
+
+        default_taxes = get_default_taxes(company)
+        company.write({"account_sale_tax_id": default_taxes[0], "account_purchase_tax_id": default_taxes[1]})
```
Comment (h4818) [inline addons/l10n_ca/models/template_ca.py:67]: single quotes and maybe split the dict keys on new lines?

File: addons/l10n_ca/data/template/account.group-ca.csv [hunk 2]
```diff
@@ -0,0 +1,169 @@
+"id","code_prefix_start","code_prefix_end","name","name@fr"
```
Comment (h4818) [inline addons/l10n_ca/data/template/account.group-ca.csv:1]: I don't see any `code_prefix_end`

**Review by h4818 — COMMENTED** [review]
Nicely done, just a few comments/thoughts...
I'm a little concerned about whether existing CA databases will upgrade, update their l10n's well.

File: addons/l10n_ca/models/res_company.py [hunk 3]
```diff
@@ -5,9 +5,11 @@
 class ResCompany(models.Model):
     _inherit = 'res.company'
 
-    l10n_ca_pst = fields.Char(related='partner_id.l10n_ca_pst', string='PST', store=False)
-
+    l10n_ca_pst = fields.Char(related='partner_id.l10n_ca_pst', string='PST', store=False, inverse='_inverse_l10n_ca_pst')
```
[the remark this answers is not in the document]
Comment (JulienAlardot) [reply addons/l10n_ca/models/res_company.py:8]: No, it doesn't change the data on the partner (and if you save it just resets to blank)
May be a bug in l10n_fr then

[the remark this answers is not in the document]
Comment (h4818) [reply addons/l10n_ca/models/res_company.py:8]: no, there is no bug in FR, and you really don't need that inverse function.

**Comment by h4818** [comment]
Also, try to remove the inheritance of `base.document.layout` -  or check with @william-andre why it is needed, and add a comment

File: addons/l10n_ca/views/res_company_view.xml [hunk 4]
```diff
@@ -7,7 +7,7 @@
             <field name="inherit_id" ref="base.view_company_form"/>
             <field name="arch" type="xml">
                 <xpath expr="//field[@name='vat']" position="after">
-                    <field name="l10n_ca_pst" attrs="{'invisible': [('country_code', '!=', 'CA')]}"/>
+                    <field name="l10n_ca_pst" attrs="{'invisible': [('country_code', '!=', 'CA')]}" readonly="False"/>
```
Comment (william-andre) [inline addons/l10n_ca/views/res_company_view.xml:10]: Is there a reason why it's not declared on the field declaration instead?

**Review by william-andre — COMMENTED** [review]
> Also, try to remove the inheritance of `base.document.layout`

Yup, I don't see why it is there (didn't blame the code)

File: addons/l10n_ca/data/template/account.account-ca.csv [hunk 5]
```diff
@@ -1,70 +1,340 @@
...
+l10n_ca_2212,221200,Other Accounts Payable and Accruals (Non-Affiliates),liability_payable,account.account_tag_financing,True,Autres comptes créditeurs et charges à payer (sociétés non affiliées)
+l10n_ca_221201,221201,Interest Payable and Accrued,liability_payable,account.account_tag_financing,True,Intérêts courus à payer
+l10n_ca_221202,221202,Amounts Payable to Members of Non-Profit Organization,liability_payable,account.account_tag_financing,True,Montants à payer aux membres d'organismes sans but lucratif
+l10n_ca_221203,221203,Wage payable and Accrued,liability_payable,account.account_tag_financing,True,Salaires courus à payer
+l10n_ca_221204,221204,Management Fee Payable and Accrued,liability_payable,account.account_tag_financing,True,Frais de gestion courus à payer
+l10n_ca_221205,221205,Bonus Payable and Accrued,liability_payable,account.account_tag_financing,True,Gratifications courues à payer
+l10n_ca_221206,221206,Employee Deduction Payable,liability_payable,account.account_tag_financing,True,Retenues salariales à payer
```
[the remark this answers is not in the document]
Comment (JulienAlardot) [reply addons/l10n_ca/data/template/account.account-ca.csv]: I was aware of that and asked a few PO's (chkl, tsb, and labe) they validated it as "there wasn't too many ones"

File: addons/l10n_ca/data/tax_report.xml [hunk 6]
```diff
@@ -0,0 +1,847 @@
...
+                <field name="expression_ids">
+                    <record id="l10n_ca_tr_gsthst_106_tag" model="account.report.expression">
+                        <field name="label">balance</field>
+                        <field name="engine">tax_tags</field>
+                        <field name="formula">Ca106</field>
+                    </record>
+                </field>
+            </record>
+            <record id="l10n_ca_tr_gsthst_107" model="account.report.line">
+                <field name="name">107 - Adjustments to be deducted when determining the net tax for the reporting period (for example, the GST/HST included in a bad debt)</field>
```
[the remark this answers is not in the document]
Comment (JulienAlardot) [reply addons/l10n_ca/data/tax_report.xml:70]: PO demand

**Pull request closed without merging**